extensions/Worker.Extensions.ServiceBus/src/ServiceBusMessageActions.cs

```diff
@@ -0,0 +1,134 @@
+        public virtual async Task CompleteMessageAsync(
+            ServiceBusReceivedMessage message,
+            CancellationToken cancellationToken = default)
+        {
+            await _settlement.CompleteAsync(new() { Locktoken = message.LockToken }, cancellationToken: cancellationToken);
+        }
+
+        ///<inheritdoc cref="ServiceBusReceiver.CompleteMessageAsync(ServiceBusReceivedMessage, CancellationToken)"/>
+        public virtual async Task AbandonMessageAsync(
+            ServiceBusReceivedMessage message,
+            IDictionary<string, object>? properties,
+            CancellationToken cancellationToken = default)
+        {
+            var request = new AbandonRequest()
+            {
+                Locktoken = message.LockToken,
+                PropertiesToModify = { TransformProperties(properties) }
+            };
+            await _settlement.AbandonAsync(request, cancellationToken: cancellationToken);
+        }
+
+        ///<inheritdoc cref="ServiceBusReceiver.DeadLetterMessageAsync(ServiceBusReceivedMessage, IDictionary{string, object}, string, string, CancellationToken)"/>
+        public virtual async Task DeadLetterMessageAsync(
+            ServiceBusReceivedMessage message,
+            Dictionary<string, object>? propertiesToModify = default,
+            string? deadLetterReason = default,
+            string? deadLetterErrorDescription = default,
+            CancellationToken cancellationToken = default)
+        {
+            var request = new DeadletterRequest()
+            {
+                Locktoken = message.LockToken,
+                PropertiesToModify = { TransformProperties(propertiesToModify) },
+                DeadletterReason = deadLetterReason,
+                DeadletterErrorDescription = deadLetterErrorDescription
+            };
+
+            await _settlement.DeadletterAsync(request, cancellationToken: cancellationToken);
+        }
+
+        ///<inheritdoc cref="ServiceBusReceiver.DeferMessageAsync(ServiceBusReceivedMessage, IDictionary{string, object}, CancellationToken)"/>
+        public virtual async Task DeferMessageAsync(
+            ServiceBusReceivedMessage message,
+            IDictionary<string, object>? propertiesToModify = default,
+            CancellationToken cancellationToken = default)
+        {
+            var request = new DeferRequest()
+            {
+                Locktoken = message.LockToken,
+                PropertiesToModify = { TransformProperties(propertiesToModify) }
+            };
+            await _settlement.DeferAsync(request, cancellationToken: cancellationToken);
+        }
+
+        private static Dictionary<string, SettlementProperties> TransformProperties(IDictionary<string, object>? properties)
+        {
+            var converted = new Dictionary<string, SettlementProperties>();
+            if (properties == null)
+            {
+                return converted;
+            }
```

Comment:
Can we modify this method to return an `IDictionary<string, SettlementProperties>` and make the following modification?

``` csharp
 if (properties == null)
 {
     return ImmutableDictionary<string, SettlementProperties>.Empty;
 }

 var converted = new Dictionary<string, SettlementProperties>();
```

Reply:
I've changed the implementation to no longer use maps in the proto file. I will update this PR shortly.